examples/streamer.rs

```diff
@@ -0,0 +1,73 @@
+use bytes::Bytes;
+use clap::{App, Arg};
+use futures_util::sink::SinkExt;
+use srt::SrtSocketBuilder;
+use std::io::Read;
+
+use std::fs::File;
+use std::time::Instant;
+
+#[tokio::main]
+async fn main() -> std::io::Result<()> {
+    let matches = App::new("srt-dump")
+        .version("1.0")
+        .about("Listen SRT stream and dump information")
+        .arg(
+            Arg::with_name("port")
+                .short("p")
+                .long("port")
+                .value_name("PORT")
+                .help("Port")
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("files")
+                .short("f")
+                .required(true)
+                .min_values(1)
+                .help("Input file for stream"),
+        )
+        .arg(
+            Arg::with_name("chunk-size")
+                .short("c")
+                .help("Chunk size")
+                .takes_value(true),
+        )
+        .get_matches();
+
+    let port = matches.value_of("port").unwrap_or("3333").to_owned();
+    print!("Value for port: {}", port);
+
+    let url = format!("127.0.0.1:{}", port);
+    let mut srt_socket = SrtSocketBuilder::new_connect(&url).connect().await.unwrap();
+
+    let chunk_size = matches
+        .value_of("chunk-size")
+        .unwrap_or("10")
+        .to_owned()
+        .parse::<u64>()
+        .unwrap();
+
+    print!("Stream hooked");
+
+    let files: Vec<_> = matches.values_of("files").unwrap().collect();
+
+    print!("{:?}", files);
+
+    loop {
+        for file in files.iter() {
+            print!("Open file {}", file);
+            let mut f = File::open(file)?;
+            let metadata = f.metadata()?;
+            let mut buffer = vec![0; chunk_size as usize];
+
+            for _ in 0..metadata.len() / chunk_size {
+                f.read_exact(&mut buffer)?;
```

Comment:
use `read()` and get the number of bytes readed.
If it's [0](https://doc.rust-lang.org/std/io/trait.Read.html#tymethod.read), it's EOF.